src/components/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,123 @@
+import { partialRight, pipe, prop } from 'ramda';
+
+import { ErrorBoundaryProps } from './ErrorBoundary.types';
+import { Inline, Stack } from '../layout';
+import { Link, Text } from '../typographyLegacy';
+import { pxToRem } from '../../utils';
+
+const IllustrationContainerSizes = {
+  sm: 24,
+  md: 48,
+  lg: 64,
+};
+
+const getIllustrationSize = pipe(
+  prop('size'),
+  partialRight(prop, [IllustrationContainerSizes]),
+  pxToRem,
+);
+
+const IllustrationContainer = styled.div<{ size: string }>`
+  width: ${getIllustrationSize};
+  height: ${getIllustrationSize};
+`;
+
+const DEFAULT_TITLE = 'We cannot show this information now';
+const DEFAULT_CONTENT = 'If the problem persists, contact support.';
+
+const ExtraSmallErrorBoundary: React.FC<ErrorBoundaryProps> = ({
+  content = 'Failed to load.',
+}) => (
+  <Text size="md" variant="secondary">
+    {content}
+  </Text>
+);
+
+const SmallErrorBoundary: React.FC<ErrorBoundaryProps> = ({
+  title = DEFAULT_TITLE,
+  content = DEFAULT_CONTENT,
+  illustration,
+  onClick,
+  size,
+}) => {
+  return (
+    <Inline align="center" gap="md">
+      {illustration && (
+        <IllustrationContainer size={size}>
+          {illustration}
+        </IllustrationContainer>
+      )}
+      <Text size="md" variant="secondary" isBold>
+        {title}
+      </Text>
+      <Text size="md" variant="secondary">
+        {content}
+      </Text>
+      {onClick && <Link onClick={onClick}>Try Again</Link>}
+    </Inline>
+  );
+};
+const MediumErrorBoundary: React.FC<ErrorBoundaryProps> = ({
+  illustration,
+  title = DEFAULT_TITLE,
+  content = DEFAULT_CONTENT,
+  justify = 'center',
+  size,
+  onClick,
+}) => {
+  return (
+    <Stack gap="sm" justify={justify}>
+      {illustration && (
+        <IllustrationContainer size={size}>
+          {illustration}
+        </IllustrationContainer>
+      )}
+      <Text size="h4" variant="secondary">
+        {title}
+      </Text>
+      <Text variant="secondary">{content}</Text>
+      {onClick && <Link onClick={onClick}>Try Again</Link>}
+    </Stack>
+  );
+};
+
+const ComponentSizeMap = {
+  xs: ExtraSmallErrorBoundary,
+  sm: SmallErrorBoundary,
+  md: MediumErrorBoundary,
+  lg: MediumErrorBoundary,
+};
+
+/* eslint-disable */
+class ErrorBoundary extends React.Component<
+  ErrorBoundaryProps,
+  { hasError: boolean }
+> {
+  constructor(props) {
+    super(props);
+    // eslint-disable-next-line
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError() {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true };
+  }
+
+  componentDidCatch(error, info) {}
+
+  render() {
+    const { size } = this.props;
+    const Component = ComponentSizeMap[size];
+    if (this?.state.hasError) {
+      return <Component {...this.props} />;
+    } else {
+      return this.props.children;
+    }
+  }
+}
+/* eslint-enable */
+
+export default ErrorBoundary;
```

Comment:
@julesbarr This should have been simple presentational component that will be used by the error boundary component that's already implemented in the platform. We don't have the ambition to implement error boundary functionality in DS